src/btree/bt_cursor.c

```diff
@@ -778,6 +832,53 @@ __wt_btcur_search(WT_CURSOR_BTREE *cbt)
     return (ret);
 }
 
+/*
+ * __btcur_search_near_bounds_reposition --
+ *     Search near with a bounded cursor should reposition the cursor at the nearest bound.
```

Comment:
I would a comment to explain the scenario why we want to reposition on lower_bound or upper_bound